### ISSUE TITLE
Params not refreshed in cellRendererFramework

### ISSUE THREAD

**Issue by britvik** [ag-grid/ag-grid#1922]
**I'm submitting a ...**
```
[X] bug report
```

**Current behavior**
params.data and params.node.data in ICellRendererAngularComp is not equal after setting new data to grid

**Expected behavior**
Params should refresh when data changes. Or agInit should be called again (like in last ag-grid version). Or have a custom callback function that will refresh the params.

**Minimal reproduction of the problem with instructions**
* Have grid with option:
  - deltaRowDataMode: true, // not sure if this option is mandatory to reproduce the bug

* Have custom column renderer:
  - cellRendererFramework: MyComponent //MyComponent implements ICellRendererAngularComp

In agInit function I'm setting various properties of params to local variables (like this.row = params.data). But when I set new data to grid (via gridOptions.api.setRowData(data)), agInit in my custom component is not called and I still have the old data in this.row.

I believe this bug was introduced by the new rendering changes.

**Please tell us about your environment:**
* **ag-Grid version:** 13.1.2
Worked in 11.x.x

* **Browser:**
Mozilla/5.0 (Windows NT 10.0; Win64; x64) AppleWebKit/537.36 (KHTML, like Gecko) Chrome/60.0.3112.113 Safari/537.36

* **Language:** [all | TypeScript X.X | ES6/7 | ES5]
all

**Comment by seanlandsman** [comment]
Hi,

Thanks for raising this. I've raised AG-826 our side to track this - you can view its progress on our pipeline page here: https://www.ag-grid.com/ag-grid-pipeline/

We'll update this thread with updates as soon as we have any.

thanks

**Comment by jbarber2016** [comment]
I have a slightly different issue when using ICellRendererAngularComp on a custom cellRenderer.  When calling gridOptions.api.updateRowData({update: updates}); the refresh function is never called.

**Comment by ceolter** [comment]
worked on this and set up and example and cannot reproduce. however i also looked at the code that is creating the params, and it is taking it from the same place, so not possible for them to be out of sync. you can see it here:
https://github.com/ag-grid/ag-grid/blob/14.1.0/src/ts/rendering/cellComp.ts#L720

the data for params comes from rowNode.data, so not possible for them to be different.

**Comment by davertron** [comment]
I'm still seeing this same issue with React and ag-grid 18.1.1:

https://codesandbox.io/s/yvp15nkny9

The top grid has no cellRendererFramework supplied, the bottom does. Notice how when right-clicking and adding items, the top grid 'samePathCount' column is correct, the bottom is only correct for new items added (unless you force re-renders by collapsing and re-expanding, etc.)

**Comment by davertron** [comment]
Just in case someone else stubs their toes on this and stumbles across this issue...

I seem to have fixed the issue by using cellRenderer in concert with frameworkComponents and a separate React component as the renderer. You can see some info and examples here https://www.ag-grid.com/javascript-grid-cell-rendering-components/#react-cell-rendering

**Comment by aruprakshit** [comment]
@davertron How did you fix it? Here is my small [demo app](https://stackblitz.com/edit/react-ag-grid-t2t-4249-shi9s9?file=src%2FGrid.js), where you see `console.log("allCounts", allCounts);` always empty. It seems the props are not updating. Do you have any idea or config that you can share?

**Comment by flatbeats** [comment]
I'm also having this issue with "ag-grid-angular": "^24.1.0","ag-grid-community": "^24.1.0",
My cellrenderer:
`@Component({
  selector: 'app-icon-cell-renderer',
  templateUrl: './icon-cell-renderer.component.html',
  styleUrls: ['./icon-cell-renderer.component.scss']
})
export class IconCellRendererComponent implements ICellRendererAngularComp {
  private params: any;
  private field: string;

  agInit(params: ICellRendererParams): void {
    this.params = params;
    this.field = params.colDef?.field;
  }

  get value(): string {
    return this.params.data[this.field];
  }

  refresh(params: any): boolean {
    return false;
  }
}
`
I solved it by manually updating the data in the cell renderer:
`  updateRow(row: SubscriptionListItem) {
    this.gridApi.applyTransaction({
      update: [row]
    });
    const iconCellRenderer = this.gridApi.getCellRendererInstances().find(
      (renderer) =>
        (renderer as any)._frameworkComponentInstance.constructor.name ===
          IconCellRendererComponent.name &&
        (renderer as any)._params.data.uuid === row.uuid
    );
    (iconCellRenderer as any)._params.data = this.gridApi.getRowNode(row.uuid).data;
  }`
Did i miss something or is this still a bug ?

**Comment by waratah** [comment]
I spent a while on this today and I am leaving a trace for others to follow.   I had a button that gets changed depending upon data.   I was using params.data to get that data and set things like the tooltip.    After a frustrating hour what I found was that the field I picked did not update therefore the refresh was never called.

Initially I was using the Id which is set and forget the record identifier.   It never changes.

I changed to the Status.  This changed from Full to Empty when I wanted the button to update so the whole refresh worked correctly.

So when coding a refresh make sure your value that you column is based upon really does change.

**Comment by BradWells** [comment]
Ran into the same issue @waratah was having with some action buttons in my app.

Confirmed that in my column definitions, changing `field: 'actions'` (a list of strings of the actions a user could preform) to `field: 'arrayMyButtonUsesThatChangesOnRefetch.length'` made the renderer props refresh on a refetch.  so my buttons started using the new data and not the old.

I also found that removing the `field` prop entirely make only the renderer of the current row refresh. I'm not sure why that is.

This strange functionality may also be related to my use of frameworkComponents, I'm seeing the warning `AG Grid: As of v27, registering components via grid property frameworkComponents is deprecated. Instead register both JavaScript AND Framework Components via the components property.`
